packages/js/onboarding/src/components/WooOnboardingTask/WooOnboardingTask.js

```diff
@@ -34,23 +35,48 @@ export const trackView = async ( taskId ) => {
 	} );
 };
 
+let experimentalVariant;
 /**
  * A Fill for adding Onboarding tasks.
  *
  * @slotFill WooOnboardingTask
  * @scope woocommerce-tasks
- * @param {Object} props    React props.
- * @param {string} props.id Task id.
+ * @param {Object} props          React props.
+ * @param {string} props.variant  The variant of the task.
+ * @param {Object} props.children React component children
+ * @param {string} props.id       Task id.
  */
-const WooOnboardingTask = ( { id, ...props } ) => {
+const WooOnboardingTask = ( { id, variant, ...props } ) => {
+	useEffect( () => {
+		if ( id === 'products' ) {
+			experimentalVariant = variant;
+		}
+	}, [ id, variant ] );
+
 	return <Fill name={ 'woocommerce_onboarding_task_' + id } { ...props } />;
 };
 
+// We need this here just in case the experiment assignment takes awhile to load, so that we don't fire trackView with a blank experimentalVariant
+// Remove all of the code in this file related to experiments and variants when the product task experiment concludes and never speak of the existence of this code to anyone
+const pollForExperimentalVariant = ( id, count ) => {
+	if ( count > 20 ) {
+		trackView( id, 'experiment_timed_out' ); // if we can't fetch experiment after 4 seconds, give up
+	} else if ( experimentalVariant ) {
+		trackView( id, experimentalVariant );
+	} else {
+		setTimeout( () => pollForExperimentalVariant( id ), 200 );
```

Comment:
I think this would require passing `count + 1`
```suggestion
		setTimeout( () => pollForExperimentalVariant( id, count + 1 ), 200 );
```

Reply:
🤦‍♂️